versions/ufo3/lib.plist.md

```diff
@@ -29,6 +29,49 @@ This defines a preferred glyph name to Postscript glyph name mapping for glyphs
 
 The mapping is stored as a dictionary with glyphs names as keys and Postscript glyph names as values. Both keys and values must be strings. The values must conform to the Postscript glyph naming specification. The dictionary may contain glyph names that are not in the font. The dictionary may not contain a key, value pair for all glyphs in the font. If a glyph's name is not defined in this mapping, the glyph's name should be used as the Postscript name.
 
+#### public.postscript.hints
+
+This key provides a mapping from glyph names to sets of PostScript hints for the
+default glyph layer. The mapping is stored as a dictionary with glyph names as
+keys and Postscript hint sets as values. If a glyph's name is missing from the
+mapping, then there are no Postscript hints for that glyph.
+
+Any mapping from glyph names to sets of PostScript hints for any layer other
+than the default layer is stored in the layerinfo.plist for the layer, using the
+same key.
+
+#### Hint Dict
+
+| key         | value type | description                                                                                                                                                                                                                                                  |
+|-------------|------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| id          | string     | Hash of glyph outlines. This is computed when the glyph is hinted. It is used to determine if the glyph outline has been changed since the glyph was hinted: if it has, then the hint dict for the glyph should be deleted. See "Hint ID Computation" below. |
```

Comment:
So if one adjusts the spacing a bit, all hints are to be deleted? Isn't there a more robust hashing algorithm. Something that encodes the proportion/relations of the nodes instead of the absolute coordinate. One first step could be to store the relative position to the previous point. That would at least take care of the spacing.

Reply:
Although the hint substitution action is tied to specific points, stem positions are not: they are only relative to each other, and to the glyph origin at (0,0). Hence, shifting the outlines by changing the LSB does mean that all the hints are invalidated. 

However, changing the advance width alone does not invalidate the hints. I would further modify the proposal by omitting the width from the start of the hash string. 

I don't see any other advantage to computing relative coordinate values before computing the hash: this requires more processing time, as the values must be converted from string to real numbers, and the relative positions calculated, while the only benefit is making the pre-hash string somewhat more compact.